java/group-layers/src/main/java/com/esri/arcgisruntime/sample/grouplayers/LayersAdapter.java

```diff
@@ -162,13 +174,41 @@ private void addChildLayer(Layer childLayer) {
 
       ((LinearLayout.LayoutParams) view.getLayoutParams()).setMarginStart(
           itemView.getResources().getDimensionPixelSize(R.dimen.adapter_item_child_margin_start));
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setOnCheckedChangeListener(null);
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setChecked(childLayer.isVisible());
-      ((CheckBox) view.findViewById(R.id.layerCheckbox)).setOnCheckedChangeListener(
-          (buttonView, isChecked) -> mOnLayerCheckedChangedListener.layerCheckedChanged(childLayer, isChecked));
-      ((TextView) view.findViewById(R.id.layerNameTextView)).setText(childLayer.getName());
-      if (mChildLayout.findViewWithTag(childLayer) == null) {
-        mChildLayout.addView(view);
+
+      // if this is an exclusive layer it will have a non-null mRadioGroup
+      if (mRadioGroup != null) {
+        // hide the checkbox that exists
+        CheckBox checkBox = view.findViewById(R.id.layerCheckbox);
+        checkBox.setVisibility(View.GONE);
+        // create a radio button or reuse an old one
+        RadioButton radioButton = view.findViewWithTag("radioButton");
+        if (radioButton == null) {
+          radioButton = new RadioButton(itemView.getContext());
+          radioButton.setTag("radioButton");
+          ((ViewGroup) view).addView(radioButton, 0);
+        }
+
+        radioButton.setOnCheckedChangeListener(null);
+        radioButton.setChecked(childLayer.isVisible());
+        radioButton.setOnCheckedChangeListener(
+            ((buttonView, isChecked) -> mOnLayerCheckedChangedListener.layerCheckedChanged(childLayer, isChecked)));
+        ((TextView) view.findViewById(R.id.layerNameTextView)).setText(childLayer.getName());
```

Comment:
get a reference to the text view first, then do some work with it. This is alot to read on one line